R/db.R

```diff
@@ -98,10 +97,10 @@ st_read_db = function(conn = NULL, table = NULL, query = NULL,
 #'   library(RPostgreSQL)
 #'   try(conn <- dbConnect(PostgreSQL(), dbname = "postgis"))
 #'   if (exists("conn") && !inherits(conn, "try-error"))
-#'     st_write_db(conn, sf, "meuse_tbl", drop = FALSE)
+#'     st_write(conn, sf, "meuse_tbl", drop = FALSE)
 #' }
-#' @details st_write_db was written with help of Josh London, see \url{https://github.com/r-spatial/sf/issues/285}
-st_write_db = function(conn = NULL, obj, table = deparse(substitute(obj)), geom_name = "wkb_geometry",
+#' @details database reading was written with help of Josh London, see \url{https://github.com/r-spatial/sf/issues/285}
+st_write.PostgreSQLConnection = function(conn = NULL, obj, table = deparse(substitute(obj)), geom_name = "wkb_geometry",
```

Comment:
the arguments before ... in the generic need to have identical names to that of the generic, which is: `obj, dsn, layer`; similar for `st_read`.

Reply:
The problem is that if we put the connection (`dsn`) in second, then we'll need to use S4 generic. Otherwise we'll only catch `st_write.sf()`. It also seems more common to see the connection as the first argument. S4?

Reply:
Better to branch in `st_write.sf` - then I'm pragmatic and would like to avoid S4. 

For `write_xxx` function it's more common to see the object to write as first argument, I'd say.